### ISSUE TITLE
Support for version in Amazon API and filenames

### ISSUE THREAD

**Issue by kilna-magento** [nficano/python-lambda#68]
Currently filenames are coded only with a timestamp...  ideally there would be provisions for creating the filename based on a version string.  The same input could then in turn be used to set the version as per the AWS Lambda API:

https://docs.aws.amazon.com/lambda/latest/dg/versioning-intro.html

This could be implemented by adding a "version: x.x.x" line to the config.yaml file, if present then the version provided will be used, otherwise the code will exhibit the current behavior for backward compatibility. Another option would be to add a '--version=x.x.x' command line parameter to 'lambda build' and the pending S3 change 'lambda upload'.  

It would break 100% compatibility with current outputs, but it would also likely be more intuitive to have the filenames generated be in the format <function name>-<version|timestamp>.zip rather than the current <timestamp>-<function name>.zip.

**Comment by kilna-magento** [comment]
Tagging my other non-work account @kilna

**Comment by kilna** [comment]
Also, bears mentioning... looking at the codebase, I can do this. Just curious if others have the same need.

**Comment by alvsanand** [comment]
Me too

**Comment by mikeycoxon** [comment]
Yes, I think this would be a very useful feature. I was thinking about submitting a patch for this along the same lines, so if you're doing it, I'll definitely support and use it.
